### DISCUSSION TITLE
Compatible rotary encoder and button librarys

### DISCUSSION THREAD

**Discussion by D34TH5TR1K3** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#300] [Q&A] [answered]
Hello,
I would like to add a five-way-switch and a rotary encoder to my esp32 that is also driving a matrix. Are there any recommented librarys for this that work well with this one? Please excuse my dumb question but it is hard for me to understand what these librarys do in the background (If I could code it myself I would not use a library lol). Just tought that maybe there are tested and proven librarys for rotary encoders and buttons + this library.
Thanks!

**Comment by mrcodetastic** [comment, marked as answer]
https://github.com/mathertel/RotaryEncoder

**Reply by D34TH5TR1K3** [reply]
Thanks! I struggled with the encoder so I am using my display now as a clock and I am not using physical input but I will probably add it in the future!
